x-pack/plugins/triggers_actions_ui/public/common/expression_items/when.tsx

```diff
@@ -65,9 +73,29 @@ export const WhenExpression = ({
     >
       <div>
         <EuiPopoverTitle>
-          {i18n.translate('xpack.triggersActionsUI.common.expressionItems.threshold.popoverTitle', {
-            defaultMessage: 'when',
-          })}
+          <EuiFlexGroup alignItems="center" gutterSize="s">
+            <EuiFlexItem>
+              {i18n.translate(
+                'xpack.triggersActionsUI.common.expressionItems.threshold.popoverTitle',
+                { defaultMessage: 'when' }
+              )}
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiButtonIcon
+                iconType="cross"
+                color="danger"
+                aria-label={i18n.translate(
+                  'xpack.triggersActionsUI.common.expressionItems.threshold.closePopoverLabel',
+                  {
+                    defaultMessage: 'Close',
+                  }
+                )}
+                onClick={() => {
+                  setAggTypePopoverOpen(false);
+                }}
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
```

Comment:
Feels like a lot of duplication which can be simplified into a single "Closable" component, no?

Seems like you can make a component which wraps around the Title on each one of these.
Something like this:

```
const ClosableTitle = ({ children }) =>
          <EuiFlexGroup alignItems="center" gutterSize="s">
            <EuiFlexItem>
            {children}
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <EuiButtonIcon
                iconType="cross"
                color="danger"
                aria-label={i18n.translate(
                  'xpack.triggersActionsUI.common.expressionItems.title.closePopoverLabel',
                  {
                    defaultMessage: 'Close',
                  }
                )}
                onClick={() => {
                  setAggTypePopoverOpen(false);
                }}
              />
            </EuiFlexItem>
          </EuiFlexGroup>
```

Then you would wrap each one of these like so:

```
<ClosableTitle>
            <EuiFlexItem>
              {i18n.translate(
                'xpack.triggersActionsUI.common.expressionItems.threshold.popoverTitle',
                { defaultMessage: 'when' }
              )}
            </EuiFlexItem>
</ClosableTitle>
```